lib/src/shared/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -390,6 +391,60 @@ public void stop() {
     @NonNull
     public Status getStatus() { return status.copy(); }
 
+    @NonNull
+    public Set<String> getPendingDocumentIds() throws CouchbaseLiteException {
+        if (config.getReplicatorType().equals(ReplicatorConfiguration.ReplicatorType.PULL)) {
+            throw new CouchbaseLiteException(
+                "Pending Document IDs are not supported on pull-only replicators.",
+                Log.LOGGING_DOMAINS_TO_C4.get(DOMAIN),
+                CBLError.Code.UNSUPPORTED);
+        }
+
+        final C4Replicator repl;
+        synchronized (lock) { repl = c4repl; }
+
+        if (repl == null) {
+            Log.i(DOMAIN, "%s: Call to getPendingDocumentIds with unstarted replicator", this);
+            return Collections.emptySet();
+        }
+
+        final Set<String> pending;
+        try { pending = c4repl.getPendingDocIDs(); }
+        catch (LiteCoreException e) {
+            Log.w(DOMAIN, "Error while fetching pending documentIds: %d/%d", e.domain, e.code);
+            return Collections.emptySet();
+        }
+
+        return (pending.size() <= 0) ? Collections.emptySet() : Collections.unmodifiableSet(pending);
+    }
+
+    @NonNull
+    public boolean isDocumentPending(@NonNull String docId) throws CouchbaseLiteException {
+        Preconditions.checkArgNotNull(docId, "document ID");
+
+        if (config.getReplicatorType().equals(ReplicatorConfiguration.ReplicatorType.PULL)) {
+            throw new CouchbaseLiteException(
+                "Pending Document IDs are not supported on pull-only replicators.",
+                Log.LOGGING_DOMAINS_TO_C4.get(DOMAIN),
+                CBLError.Code.UNSUPPORTED);
+        }
+
+        final C4Replicator repl;
+        synchronized (lock) { repl = c4repl; }
+
+        if (repl == null) {
```

Comment:
Per requirement, we should be able to return the result here as well. From the slack discussion with @borrrden, we will need to recreate c4repl here if it's null. This will also need to change logic in some other place that check null for c4repl as well.